cpp/lib/src/util.cc

```diff
@@ -25,15 +25,37 @@
 #include <stdexcept>
 #include <cctype>
 #include <cerrno>
+#include <csignal>
 #include <cstdlib>
 #include <execinfo.h>
 #include <signal.h>
 #include "Compiler.h"
+#include "FileLocker.h"
+#include "FileUtil.h"
 #include "MiscUtil.h"
+#include "SignalUtil.h"
 #include "StringUtil.h"
 #include "TimeUtil.h"
 
 
+static std::string log_path;
+static int log_fd;
+
+
+void HUPHandler(int /* signal_number*/) {
+    ::close(log_fd);
+    const int new_fd(::open(log_path.c_str(), O_WRONLY|O_APPEND));
+    if (new_fd == -1) { // We're screwed!
+        const std::string error_message("in HUPHandler(util.cc): can't open(2) \"" + log_path + "\"!\n");
+        ::write(STDERR_FILENO, error_message.c_str(), error_message.size());
+        ::exit(EXIT_FAILURE);
+    }
+
+    // Make sure we're using the same file descriptor as before!
+    ::dup2(new_fd, log_fd);
```

Comment:
what if ::dup2 fails?

Reply:
and what happens with fd_? maybe just call redirectOutput(new_fd) for consistency?